c/python/args.cc

```diff
@@ -39,35 +63,114 @@ Arg Args::get(const char*) const { throw AssertionError(); }
 
 void NoArgs::bind(PyObject* _args, PyObject* _kwds) {
   if ((_args && Py_SIZE(_args)) || (_kwds && PyDict_Size(_kwds))) {
-    throw TypeError() << fn_name << "() accepts no arguments";
+    throw TypeError() << get_name() << " accepts no arguments";
   }
 }
 
 
 
 //------------------------------------------------------------------------------
-// PosAndKwdArgs
+// PKArgs
 //------------------------------------------------------------------------------
 
-PosAndKwdArgs::PosAndKwdArgs(
-    const char* name, size_t np, std::initializer_list<const char*> ki)
-    : Args(name), n_posonly_args(np), n_kwdonly_args(0), kwd_names(ki),
-      args(nullptr), kwds(nullptr) {}
+PKArgs::PKArgs(
+    size_t npo, size_t npk, size_t nko, bool vargs, bool vkwds,
+    std::initializer_list<const char*> _names
+  )
+  : n_posonly_args(npo),
+    n_pos_kwd_args(npk),
+    n_kwdonly_args(nko),
+    n_args(npo + npk + nko),
+    has_varargs(vargs),
+    has_varkwds(vkwds),
+    arg_names(_names)
+{
+  xassert(n_args == arg_names.size());
+  if (has_varargs) xassert(n_pos_kwd_args == 0);
+  bound_args.resize(n_args);
+  for (size_t i = 0; i < n_args; ++i) {
+    bound_args[i].init(i, this);
+  }
+}
```

Comment:
`n_kwdonly_args` and `has_varkwds` are not used for the moment, right?

Reply:
That's right, as the warning message will keep pestering you about:
```
c/python/args.h:73:18: warning: private field 'n_kwdonly_args' is not used [-Wunused-private-field]
    const size_t n_kwdonly_args;
                 ^
c/python/args.h:76:18: warning: private field 'has_varkwds' is not used [-Wunused-private-field]
    const bool   has_varkwds;
                 ^
2 warnings generated.
```